### PR TITLE
[P-2-4-58] Add solutions written in: Python, JavaScript and OCaml

### PR DESCRIPTION
**Pull request by A-Hilaly**
Added js solutions and P058 in python

@ryukinix 

I would like to know what the project is focused on, adding new solutions (to unsolved problems) or language diversity. so I consider it my next pull requests !

### CODE REVIEW

**Comment by rafaelcn** [comment]
Hello and thanks for the contributions! 
We are currently focused on both: solutions and language diversity. There is no priority on which one, though. Send us many contributions as you have 😆

**Comment by A-Hilaly** [comment]
Alright, going too ! 👍

**Comment by ryukinix** [comment]
Thanks for the contribution, @A-Hilaly.

Please, review the Python solution for the Problem058 to not break the one-minute rule.  Seems that solutions takes too time to finish as I saw on travis output:

image


Another thing: try create more descriptive commit summaries for your commits, talking about the problem(s) added and the language too.

**Comment by ryukinix** [comment]
On your machine this solutions takes how much time to finish? If only few seconds more than one minute you can add this solution but with one condition: rename to `solution_slow_1.py`

**Comment by A-Hilaly** [comment]
@ryukinix Uhmm my machine took +- 3 seconds to solve the python solution

rt

I believe that they might be better solution for this one . Im trying to fix this

**Comment by A-Hilaly** [comment]
Do you have any hard info about Travis Machine ?

File: Problem058/python/solution_slow_1.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import math
```
Comment (ryukinix) [inline Problem058/python/solution_slow_1.py]: Please, fix the directory case from `python` to `Python`.

**Comment by ryukinix** [comment]
@A-Hilaly actually all have is [this](https://docs.travis-ci.com/user/reference/overview/).

But which python version are you using? On our dockerfile container, which runs on the travis build system, we are using `Python3.4`.

File: Problem058/python/solution_slow_1.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+    @classmethod
+    def solve(cls):
+        obj = object.__new__(cls)
+        obj.__init__()
+        return obj._solve(0.1)
+
+
+if __name__ == "__main__":
+    print(Problem058.solve())
```
Comment (ryukinix) [inline Problem058/python/solution_slow_1.py]: You can return only the answer instead a pair of numbers? That way we can easilly check for the answer later with `stats.py`.

**Comment by ryukinix** [comment]
Your code for Problem058 seems fine, @A-Hilaly. I didn't understand why takes so long to finish on travis. I'll make some manual tests here.

**Comment by ryukinix** [comment]
@A-Hilaly tested here, on my machine the solution takes about 3.8s and this time time is pretty fine. 

image

I'm wondering why take so long to finish on travis...

**Comment by A-Hilaly** [comment]
Im using last python version, but I don't think syntax is different from 3.X python versions

**Comment by ryukinix** [comment]
Ok. I'm executed using the docker container and the result is the same: runs on about 3.4s.

**Comment by ryukinix** [comment]
@A-Hilaly your PR is just fine for now. I'll just do a little rebase to fix this mess-up git messages, ok? Wait a little and avoid to push new things to that.

**Comment by A-Hilaly** [comment]
Sure ! I saw travis last build its taking 13seconds, this is what you got yesterday ?
p58

**Comment by ryukinix** [comment]
Actually yesterday the build just locked in something like to a infinite loop and after 10 minutes the build was killed. The last message seems that the `Elixir` build was executing the Problem058 solution. Well, let enable again the solution_1.py to build and let's see what happens. Seems for some reason fixed now. Maybe was just a random problem on travis?

**Comment by A-Hilaly** [comment]
It is weird, the only thing different from yesterday is the readme file 😆 . I never did some elixir before but I understand thats it running solutions without checking the output right ?

**Comment by ryukinix** [comment]
@A-Hilaly now just wait and since the test pass I'll be merging your PR :)

PS.: please, as I made a rebase, if you want to make a new commit, refresh our branch by cloning the repository again or using another technique.

**Comment by ryukinix** [comment]
Yes, the only script which check the output is the Python version (`stats.py`). The Elixir version is a asynchronous version of `stats.py` to build more fast when multiple cores are avaible. Since that we cut time of tests from 10min (yes.. pretty slow) to 4min using Docker + Asynchronous build.

**Comment by ryukinix** [comment]
image

Yes, your solution is passing on tests now, for some reason. Great. I'll merge now. Thanks for the patience on review!

**Comment by A-Hilaly** [comment]
@ryukinix Okkey thanks ! Ill pay attention to refresh forks stuff next time :)

**Pull request merged**